iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/planner/plan/node/process/ActiveRegionScanMergeNode.java

```diff
@@ -70,13 +73,20 @@ public PlanNode clone() {
 
   @Override
   public List<String> getOutputColumnNames() {
-    return outputCount
-        ? ColumnHeaderConstant.countDevicesColumnHeaders.stream()
-            .map(ColumnHeader::getColumnName)
-            .collect(Collectors.toList())
-        : ColumnHeaderConstant.showDevicesColumnHeaders.stream()
-            .map(ColumnHeader::getColumnName)
-            .collect(Collectors.toList());
+    if (!outputCount) {
+      return children.get(0).getOutputColumnNames();
+    }
+    if (children.get(0) instanceof DeviceRegionScanNode) {
+      return ColumnHeaderConstant.showDevicesColumnHeaders.stream()
+          .map(ColumnHeader::getColumnName)
+          .collect(Collectors.toList());
+    } else if (children.get(0) instanceof TimeSeriesSchemaScanNode) {
+      return ColumnHeaderConstant.showTimeSeriesColumnHeaders.stream()
+          .map(ColumnHeader::getColumnName)
+          .collect(Collectors.toList());
```

Comment:
If needMerge is false, we can also output the columnNames of child directly. Because the form of output from child is COUNT already.

Reply:
Is this what you mean?
```suggestion
    if (!outputCount|| !needMerge) {
      return children.get(0).getOutputColumnNames();
    }
    if (children.get(0) instanceof DeviceRegionScanNode) {
      return ColumnHeaderConstant.showDevicesColumnHeaders.stream()
          .map(ColumnHeader::getColumnName)
          .collect(Collectors.toList());
    } else if (children.get(0) instanceof TimeSeriesSchemaScanNode) {
      return ColumnHeaderConstant.showTimeSeriesColumnHeaders.stream()
          .map(ColumnHeader::getColumnName)
          .collect(Collectors.toList());
```

Reply:
> Is this what you mean?
✅